bolt-contracts/src/interfaces/IBoltMiddleware.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import {BLS12381} from "../lib/bls/BLS12381.sol";
+import {IBoltValidators} from "./IBoltValidators.sol";
+import {IBoltManager} from "./IBoltManager.sol";
+
+interface IBoltMiddleware {
+    error InvalidQuery();
+    error AlreadyRegistered();
+    error NotRegistered();
+    error OperatorNotOptedIn();
+    error NotOperator();
+    error CollateralNotWhitelisted();
+    error NotAllowed();
+
+    function nameHash() external view returns (bytes32);
+
+    function getEpochStartTs(
+        uint48 epoch
+    ) external view returns (uint48);
+
+    function getEpochAtTs(
+        uint48 timestamp
+    ) external view returns (uint48);
+
+    function getCurrentEpoch() external view returns (uint48);
+
+    function addWhitelistedCollateral(
+        address collateral
+    ) external;
+
+    function removeWhitelistedCollateral(
+        address collateral
+    ) external;
```

Comment:
why is this part of the middleware? IMO handling collateral whitelist should be a responsibility of the manager governance, wdyt?